docs/configuration-discovery-advanced.md

```diff
@@ -0,0 +1,79 @@
+# Configuration: advanced discovery
+
+The Orchestrator maintains an internal queue of instances to discover. When the instance is scheduled to be discovered, it is added to the queue.
```

Comment:
The Orchestrator uses an internal queue to manage instances for discovery. When an instance is ready for discovery, it gets added to the queue. Discovery workers process the queue. The `DiscoveryMaxConcurrency` setting in a configuration file controls the number of workers. This setting determines how many discoveries can happen in parallel.